File: clientwebsite/src/app/modules/shared/models/notification.ts
import { Poll } from '../../company-dashboard/models/poll.model';
import { NotificationType } from './notification-type';
import { Task } from './task';
import { UserDetails } from './user-details';

export interface Notification {
  id: number;
  date: string;
  dateTime: string;
  title: string;
  body: string;
  isRead: boolean;
  isDeleted: boolean;
  userId: string;
  user: UserDetails;
  notificationTypeId: number;
  notificationType: NotificationType;
  tasksId: number;
  tasks: Task;
  pollId: number;
  poll: Poll;
}

export interface NotificationsResponse {
  notifications: Notification[];
  unReadedCount: number;
  readedCount: number;
}

File: API/Entities/Notification.cs
using System;
using API.Entities.Polls;

namespace API.Entities
{
    public class Notification
    {
        public int Id { get; set; }
        public DateTime Date { get; set; }
        public string Title { get; set; }
        public string Body { get; set; }
        public bool IsRead { get; set; }
        public bool IsDeleted { get; set; }
        public string UserId { get; set; }
        public AppUser User { get; set; }
        public string RelatedUserId { get; set; }
        public AppUser RelatedUser { get; set; }
        public int NotificationTypeId { get; set; }
        public NotificationType NotificationType { get; set; }
        public int? TasksId { get; set; }
        public Tasks Tasks { get; set; }
        public int? PollId { get; set; }
        public Poll Poll { get; set; }
    }
}